scripts/paper/generate_tablero_scaling_law.py

```
@@ -0,0 +1,382 @@
+    parts.append("</svg>")
+    path.parent.mkdir(parents=True, exist_ok=True)
+    path.write_text("\n".join(parts) + "\n", encoding="utf-8")
+
+
+def generate(inputs: List[Path], out_tsv: Path, out_json: Path, out_svg: Path) -> dict[str, object]:
+    all_points: List[CurvePoint] = []
+    for path in inputs:
+        all_points.extend(parse_input(path))
+    families = {point.family for point in all_points}
+    if families != {"default", "2x2", "3x3"}:
+        raise SystemExit(f"expected default, 2x2, and 3x3 families, got {sorted(families)}")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)